duplicate-post-admin.php

```diff
@@ -450,6 +452,65 @@ function duplicate_post_admin_enqueue_scripts( $hook ) {
 	}
 }
 
+/**
+ * Flattens a version number for use in a filename.
+ *
+ * @param string $version The original version number.
+ *
+ * @return string The flattened version number.
+ */
+function duplicate_post_flatten_version( $version ) {
+	$parts = explode( '.', $version );
+
+	if ( count( $parts ) === 2 && preg_match( '/^\d+$/', $parts[1] ) === 1 ) {
+		$parts[] = '0';
+	}
+
+	return implode( '', $parts );
+}
+
+/**
+ * Enqueues the necessary JavaScript code for the block editor.
+ *
+ * @return void
+ */
+function duplicate_post_admin_enqueue_block_editor_scripts() {
+	wp_enqueue_script(
+		'duplicate_post_edit_script',
+		plugins_url( sprintf( 'js/dist/duplicate-post-edit-%s.js', duplicate_post_flatten_version( DUPLICATE_POST_CURRENT_VERSION ) ), __FILE__ ),
+		array(
+			'wp-blocks',
+			'wp-element',
+			'wp-i18n',
+		),
+		DUPLICATE_POST_CURRENT_VERSION,
+		true
+	);
+
+	wp_localize_script(
+		'duplicate_post_edit_script',
+		'duplicatePostRewriteRepost',
+		array(
+			'permalink' => duplicate_post_get_rewrite_republish_permalink(),
```

Comment:
Not necessarily in the scope of this PR. Currently, this passes the link with the action parameter to create a new draft. Will need to be changed to an action specific to rewrite and republish. See similar notes below.